## Loading Dataset


In [2]:
import pandas as pd
column_names = ["iD", "entity", "sentiment", "text"]

df = pd.read_csv("twitter_training.csv",names=column_names)
print(df.head())
print(df.tail())

     iD       entity sentiment  \
0  2401  Borderlands  Positive   
1  2401  Borderlands  Positive   
2  2401  Borderlands  Positive   
3  2401  Borderlands  Positive   
4  2401  Borderlands  Positive   

                                                text  
0  im getting on borderlands and i will murder yo...  
1  I am coming to the borders and I will kill you...  
2  im getting on borderlands and i will kill you ...  
3  im coming on borderlands and i will murder you...  
4  im getting on borderlands 2 and i will murder ...  
         iD  entity sentiment  \
74677  9200  Nvidia  Positive   
74678  9200  Nvidia  Positive   
74679  9200  Nvidia  Positive   
74680  9200  Nvidia  Positive   
74681  9200  Nvidia  Positive   

                                                    text  
74677  Just realized that the Windows partition of my...  
74678  Just realized that my Mac window partition is ...  
74679  Just realized the windows partition of my Mac ...  
74680  Just realized between t

## Data Cleaning 


### Information of the data set at it's raw state

In [3]:
print("The overall information about dataframe is as follows")
print(df.info())  # To see the overall information of the data set
print("The shape of the dataframe is", df.shape)  # To check the shape of teh dataframe (Nrow,Nccolumns)

print("total number of duplicated rows,",df.duplicated().sum()) # Total Number of duplicated values
print("Null value distribution \n",df.isna().sum()) # To know the total number of null values

The overall information about dataframe is as follows
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74682 entries, 0 to 74681
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   iD         74682 non-null  int64 
 1   entity     74682 non-null  object
 2   sentiment  74682 non-null  object
 3   text       73996 non-null  object
dtypes: int64(1), object(3)
memory usage: 2.3+ MB
None
The shape of the dataframe is (74682, 4)
total number of duplicated rows, 2700
Null value distribution 
 iD             0
entity         0
sentiment      0
text         686
dtype: int64


### Data cleaning
-As we can we observe from the out put there are about 2700 duplicated rows
- And the text column has 686 Null Values
- We have to drop/remove them.


In [4]:
df= df.dropna()   # Drop rows with any missing values
df= df.drop_duplicates()  # Drop duplicate rows



### Check

In [5]:
print("The overall information about dataframe is as follows")
print(df.info())  # To see the overall information of the data set
print("The shape of the dataframe is", df.shape)  # To check the shape of teh dataframe (Nrow,Nccolumns)

print("total number of duplicated rows,",df.duplicated().sum()) # Total Number of duplicated values
print("Null value distribution \n",df.isna().sum()) # To know the total number of null values

The overall information about dataframe is as follows
<class 'pandas.core.frame.DataFrame'>
Index: 71656 entries, 0 to 74681
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   iD         71656 non-null  int64 
 1   entity     71656 non-null  object
 2   sentiment  71656 non-null  object
 3   text       71656 non-null  object
dtypes: int64(1), object(3)
memory usage: 2.7+ MB
None
The shape of the dataframe is (71656, 4)
total number of duplicated rows, 0
Null value distribution 
 iD           0
entity       0
sentiment    0
text         0
dtype: int64


### Removing Stop words

In [6]:
from nltk.corpus import stopwords
stop_words = set(stopwords.words('english'))

# Remove stopwords
df['text'] = df['text'].apply(lambda x: ' '.join([word for word in str(x).split() if word not in stop_words]))

## Wrangle Function

In [7]:
def wrangle(df_path):
    
    column_names = ["iD", "entity", "sentiment", "text"]
    df = pd.read_csv(df_path,names=column_names)

    df = pd.read_csv("twitter_training.csv",names=column_names) # Reading csv file
    df= df.dropna()   # Drop rows with any missing values
    df= df.drop_duplicates()  # Drop duplicate rows
    # Remove stopwords
    stop_words = set(stopwords.words('english'))
    df['text'] = df['text'].apply(lambda x: ' '.join([word for word in str(x).split() if word not in stop_words]))
    return(df)

In [8]:

df = wrangle("twitter_training.csv")
df.head()

iD       entity sentiment                                  text
0  2401  Borderlands  Positive       im getting borderlands murder ,
1  2401  Borderlands  Positive          I coming borders I kill all,
2  2401  Borderlands  Positive      im getting borderlands kill all,
3  2401  Borderlands  Positive     im coming borderlands murder all,
4  2401  Borderlands  Positive  im getting borderlands 2 murder all,